optuna/study/study.py

```diff
@@ -868,10 +854,6 @@ def objective(trial):
         Args:
             trial: Trial to add.
 
-        Raises:
-            :exc:`ValueError`:
-                If trial is an invalid state.
```

Comment:
What is a definition of its validity? Should we point the definition?

Reply:
I'm not sure if we have to add the definition here. The error should be raised in `create_trial` if the given state is invalid (i.e. state not in TrialState).

`add_trial` only raises an exception iff user tries to invoke `add_trial` with constructing a frozen trial by hand, which is not intended use-case (we assume that a trial is created by `create_trial` that invoke a validation).

Do you think it'd be nice to have a definition of valid states (or reference to it) here as well?

Reply:
We discussed this privately and agreed to simply delete it here. On a separate topic, we concluded that we should discuss whether to make trial validity available to users.